src/components/structures/ThreadView.tsx

```diff
@@ -241,29 +243,35 @@ export default class ThreadView extends React.Component<IProps, IState> {
         }
     };
 
+    private nextBatch: string;
+
     private onPaginationRequest = async (
         timelineWindow: TimelineWindow | null,
         direction = Direction.Backward,
         limit = 20,
     ): Promise<boolean> => {
         if (!Thread.hasServerSideSupport) {
-            return false;
+            timelineWindow.extend(direction, limit);
+            return true;
```

Comment:
I wonder if we can switch this back to `return timelineWindow.extend(direction, limit);` now that `Room.timeline` events re-set the canPaginate* states

Reply:
yup, i'll test this out